### PR TITLE
✨🏷️ 257 - Data Mapper - GET flow for Consent Release Data 

### PR DESCRIPTION
**Pull request by anncatton**
## Description of Changes

Implements Data Mapper `GET` router endpoint for `ConsentReleaseData` step of the Wizard, including steps for retrieving PI Participant and Consent ParticipantResponse data.

### Data Mapper
- adds `GET /consent-release-data` endpoint
- updates swagger schemas
- adds `getConsentReleaseData` to search service. Adds outline for all steps, and implements functions for retrieving PI and Consent DAS data
- minor name change for `createConsentReleaseData` req/res types, adds `Create` verb to differentiate from `Get` schema type
- renames `/das/consent` `getParticipantById` -> `getConsentParticipantById` to differentiate from the `/das/pi` `getPIParticipantById`

### PI DAS
- implements `GET /participants/{participantId}` endpoint
- adds `getParticipantById` search service function

### Types
- adds `GetConsentReleaseData` req/response schema types, and renames `ConsentReleaseData` request and response types for `POST` to `CreateConsentReleaseData` to differentiate the 2 types  
- adds supporting response types for PI GetParticipant and Consent `CONSENT_RELEASE_DATA` category ParticipantResponses.

### Documentation
- adds sequence diagram for GET Consent Release Data flow
- improves and renames sequence diagram for POST Consent Release Data flow

## PR Readiness Checklist

- [x] "Expected Outcome(s)" in ticket have been met
- [x] Ticket number included in PR title
- [x] Connected ticket to PR
- [x] Labels added to PR for service name (`consent-api`, `data-mapper`, etc...), type (`chore`, `documentation`, etc...), status (`draft`, `on-hold`, etc...) **if applicable**
- [x] Manual testing completed
- [x] Builds locally without errors or warnings
- [ ] Tests are updated (if required) and passing
- [ ] PR feedback has been addressed **if applicable**
- [ ] New environment variables added to `.env.schema` files, `README.md`
- [x] Added copyrights to new files

### CODE REVIEW

File: apps/data-mapper/src/services/create.ts [hunk 1]
```diff
@@ -20,8 +20,8 @@
 import {
 	ClinicianInviteRequest,
 	ClinicianInviteResponse,
-	ConsentReleaseDataRequest,
-	ConsentReleaseDataResponse,
+	CreateConsentReleaseDataRequest,
```
Comment (anncatton) [inline apps/data-mapper/src/services/create.ts:23]: Just some renaming to differentiate this request type from the `Get` type

File: apps/data-mapper/src/services/das/consent.ts [hunk 2]
```diff
@@ -307,7 +307,7 @@ type GetParticipantFailureStatus = SystemError | 'PARTICIPANT_DOES_NOT_EXIST';
  * @param participantId string
  * @returns ConsentParticipant
  */
-export const getParticipantById = async (
+export const getConsentParticipantById = async (
```
Comment (anncatton) [inline apps/data-mapper/src/services/das/consent.ts:310]: Renaming to differentiate from `PI` `getParticipantById`

File: apps/data-mapper/src/services/search.ts [hunk 3]
```diff
@@ -228,6 +230,80 @@ export const getInformedConsentResponse = async (
...
+		//** Keys DAS + PHI DAS service call for fetching Ohip data */
+		//** Keys DAS service call for fetching ClinicalProfile private key */
+		//** PHI DAS service call for ClinicalProfile */
+
+		// combine all response data and parse
+		const completeResponse = {
+			...piParticipantData.data,
+			...consentQuestionResponses.data,
+			// ...(ohipRecordResponse?.data || {}),
+			// ...updateClinicalProfileResponse.data,
```
Comment (anncatton) [inline apps/data-mapper/src/services/search.ts:290]: added in next 2 followup PRs

File: apps/data-mapper/src/services/search.ts [hunk 4]
```diff
@@ -228,6 +230,80 @@ export const getInformedConsentResponse = async (
...
+		};
+		const parsedCompleteResponse = GetConsentReleaseDataResponse.safeParse(completeResponse);
```
Comment (anncatton) [inline apps/data-mapper/src/services/search.ts:292]: This parsing of course will not succeed until the ohip and clinical profile steps are implemented

File: docs/diagrams/create-consent-release-data-flow.md [hunk 5]
```diff
@@ -20,13 +20,14 @@ sequenceDiagram
...
     mapper->>consent: POST to /participant-responses {RELEASE_DATA__DE_IDENTIFIED}
     consent->>-mapper: Consent DAS participantResponse response
-    Note over mapper: If hasOhip=true, ohipPrivateKey entry required in KEYS DAS, If hasOhip=false, proceed to clinicalProfilePrivateKey
-    mapper->>+keys: POST to /ohip-keys/{participantId}
-    Note over keys: If an ohipKey exists, it is returned. If not, a new one is created
-    keys->>-mapper: Response with ohipKey
-    mapper->>+phi: POST to /ohip/{ohipPrivateKey}
-    Note over phi: With retrieved ohipPrivateKey, update record. Use UPSERT to create if record doesn't exist.
-    phi->>-mapper: PHI DAS createOhip response
+    alt hasOhip=true
```
Comment (anncatton) [inline docs/diagrams/create-consent-release-data-flow.md:23]: updated this diagram to make use of the `alt` feature in mermaid https://mermaid.js.org/syntax/sequenceDiagram.html#alt

File: packages/types/src/services/dataMapper/requests/CreateConsentReleaseData.ts [hunk 6]
```diff
@@ -32,7 +32,7 @@ import {
 	OhipInfo,
 } from '../../../entities/ConsentReleaseData.js';
 
-export const ConsentReleaseDataRequest = ConsentReleaseDataBase.merge(
+export const CreateConsentReleaseDataRequest = ConsentReleaseDataBase.merge(
```
Comment (anncatton) [inline packages/types/src/services/dataMapper/requests/CreateConsentReleaseData.ts:35]: renaming for clarity, as mentioned above

File: apps/data-mapper/src/services/search.ts [hunk 7]
```diff
@@ -228,6 +230,80 @@ export const getInformedConsentResponse = async (
...
+	 * Steps:
+	 * 1) PI DAS - Retrieve PI Participant data
+	 * 2) Consent DAS - Retrieve most recent ParticipantResponse data for CONSENT_RELEASE_DATA ConsentCategory.
+	 * 3) If {hasOhip: true}, retrieve ohipPrivateKey:
+	 * 		a) Keys DAS - Attempt to GET ohipPrivateKey by participantId. If key is found, proceed to b); otherwise proceed to 4)
+	 * 		b) PHI DAS - With retrieved ohipPrivateKey, retrieve OhipNumber record.
+	 * 4) Retrieve clinicalProfilePrivateKey:
+	 * 		a) Keys DAS - Attempt to GET clinicalProfilePrivateKey by participantId. If key is found, proceed to b)
+	 * 		b) PHI DAS - With retrieved clinicalProfilePrivateKey, retrieve clinicalProfile.
+	 * 5) If success, return success(ConsentReleaseDataResponse). If failure at any step, return that failure()
```
Comment (mistryrn) [inline apps/data-mapper/src/services/search.ts:252]: This is super helpful! 🙇

File: apps/pi-das/src/routers/participants.ts [hunk 8]
```diff
@@ -123,20 +122,46 @@ router.get('/', async (req, res) => {
...
+		const validatedParticipantId = NanoId.safeParse(participantId);
+		if (!validatedParticipantId.success) {
+			logger.error(
+				`GET ${ROUTER_PATH}/:participantId`,
+				'Received invalid participantId',
+				validatedParticipantId.error,
+			);
+			return res.status(400).json(RequestValidationErrorResponse(validatedParticipantId.error));
+		}
+		const participant = await getParticipantById(participantId);
```
Comment (mistryrn) [inline apps/pi-das/src/routers/participants.ts:149]: 🎉

File: docs/diagrams/create-consent-release-data-flow.md (hunk 5 above)
Comment (mistryrn) [reply docs/diagrams/create-consent-release-data-flow.md:23]: This is sweet!

**Review by mistryrn — APPROVED** [review]
Looks good!

**Review by mistryrn — APPROVED** [review]
🚀

**Pull request merged**